client/my-sites/plans-grid/components/storage-add-on-dropdown.tsx

```diff
@@ -52,57 +67,79 @@ export const StorageAddOnDropdown = ( {
 	label = '',
 	planSlug,
 	storageOptions,
-	showPrice = false,
 	onStorageAddOnClick,
+	priceOnSeparateLine = false,
 }: StorageAddOnDropdownProps ) => {
+	const translate = useTranslate();
 	const { gridPlansIndex } = usePlansGridContext();
-	const { storageAddOnsForPlan } = gridPlansIndex[ planSlug ];
+	const {
+		pricing: { currencyCode },
+		storageAddOnsForPlan,
+	} = gridPlansIndex[ planSlug ];
 	const { setSelectedStorageOptionForPlan } = useDispatch( WpcomPlansUI.store );
+	const storageAddOnPrices = storageAddOnsForPlan?.map(
```

Comment:
Let's wrap this with `useMemo` to allow consumers to optimize. it's a different object now every time this function runs, irrespective of `storageAddOnsForPlan`

Reply:
Thanks -- will address this shortly 👍

Reply:
Addressed in [f3c0718](https://github.com/Automattic/wp-calypso/pull/81743/commits/f3c07188f671664c4acff3928685e77ac3d4f011)